packages/mobile/src/verify/saga.ts

```diff
@@ -685,34 +693,38 @@ export function* receiveAttestationCodeSaga(action: ReturnType<typeof receiveAtt
     action.payload.index
   )
 
-  const attestationInputStatus = yield select(attestationInputStatusSelector)
-  const index = action.payload.index ?? indexReadyForInput(attestationInputStatus)
-  if (index >= NUM_ATTESTATIONS_REQUIRED) {
-    Logger.error(
-      TAG + '@attestationCodeReceiver',
-      'All attestation code positions are full. Ignoring.'
-    )
-    return
-  }
-  yield put(setAttestationInputStatus({ index, status: CodeInputStatus.Received }))
+  // Only one code should be being received at the same time so that it doesn't cause any UI glitches by using
+  // the same index position for more that one. We use the lock for that.
+  yield call([receiveAttestationCodeLock, receiveAttestationCodeLock.acquireAsync])
```

Comment:
The diff here is a bit hard to read but all I did is move the `try` of the `try/catch` earlier in the code and put a lock (`receiveAttestationCodeLock`) so that only one code goes through this part at the time.